main/lsp/LSPIndexer.cc

```diff
@@ -82,8 +82,8 @@ void LSPIndexer::computeFileHashes(const vector<shared_ptr<core::File>> &files)
     computeFileHashes(files, *emptyWorkers);
 }
 
-bool LSPIndexer::canTakeFastPath(const std::vector<std::shared_ptr<core::File>> &changedFiles,
-                                 bool containsPendingTypecheckUpdates) const {
+bool LSPIndexer::canTakeFastPath(const vector<shared_ptr<core::File>> &changedFiles,
+                                 const UnorderedMap<core::FileRef, shared_ptr<core::File>> &evictedFiles) const {
```

Comment:
I generalized an existing mechanism which supported checking if an edit can take a fast path in the immediate past. The boolean becomes a map containing files that have been replaced in GlobalState so that we can fetch their contents.